File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    private const int CarsProductionForSpeedMeasurement = 221;
    private const int MinutesPerHour = 60;

    private const double FiveToEightCarSuccessRate = 0.9;
    private const double NineCarSuccessRate = 0.8;
    private const double TenCarSuccessRate = 0.77;

    private static int RawProductionRate(int speed) => CarsProductionForSpeedMeasurement * speed;

    public static double ProductionRatePerHour(int speed)
    {
        if (speed == 0) return 0;
        else if (speed < 5) return RawProductionRate(speed);
        else if(speed < 9) return RawProductionRate(speed) * FiveToEightCarSuccessRate;
        else if(speed < 10) return RawProductionRate(speed) * NineCarSuccessRate;
        else return RawProductionRate(speed) * TenCarSuccessRate;
    }

    public static int WorkingItemsPerMinute(int speed) => (int)ProductionRatePerHour(speed) / MinutesPerHour;
}

File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

[<Literal>]
let CarsProductedPerHour = 221
[<Literal>]
let MinutesInHour = 60

let successRate = function
| 0 -> 0.0
| s when s < 5 -> 1.0
| s when s < 9 -> 0.9 
| 9 -> 0.8
| 10 -> 0.77
| _ -> failwith "Invalid speed, needs to be between 0 and 10" 

let productionRatePerHour (speed: int): float =
    float speed * float CarsProductedPerHour * successRate speed

let workingItemsPerMinute (speed: int): int =
    int (productionRatePerHour speed) / MinutesInHour

